drivers/flash/Kconfig.mcux

```diff
@@ -36,8 +36,7 @@ menu "Flexspi flash driver"
 
 config FLASH_MCUX_FLEXSPI_NOR
 	bool "MCUX FlexSPI NOR driver"
-	default y
-	depends on DT_HAS_NXP_IMX_FLEXSPI_NOR_ENABLED
+	default y if DT_HAS_NXP_IMX_FLEXSPI_NOR_ENABLED
```

Comment:
this change doesn't make sense, previously it would be enabled by default if the node exists but could be disabled but could not be enabled if it doesn't exist, now it can be enabled if the node doesn't exist and cause a build error?

Reply:
I'm not sure what you are talking about, as far as I am aware this has the same meaning

Reply:
okay, well I;ll change it back, I think I see what you mean now